blockstore/src/main/java/brooklyn/location/blockstore/BlockDeviceOptions.java

```diff
@@ -27,6 +27,11 @@ public static BlockDeviceOptions fromMap(Map<String, ?> map) {
             }
         }
         if (map.containsKey("sizeInGb")) {
+            if (map.get("sizeInGb") instanceof Double && Math.abs((Double) map.get("sizeInGb") - ((Double) map.get("sizeInGb")).intValue()) >= 0.01
+                    || map.get("sizeInGb") instanceof Float && Math.abs((Float) map.get("sizeInGb") - ((Float) map.get("sizeInGb")).intValue()) >= 0.01) {
+                throw new UnsupportedOperationException("Trying to set block device with not allowed sizeInGb value "
+                        + map.get("sizeInGb") + "; sizeInGb must have integer value.");
+            }
             result.sizeInGb = (Integer) checkNotNull(map.get("sizeInGb"), "sizeInGb");
```

Comment:
If this is a `Double` you'll get a `ClassCastException`. You can convert to an `int` using casting to a `Number` and calling `intValue`.